src/storage/mod.rs

```diff
@@ -1040,6 +1163,106 @@ impl Storage {
                         &Key::from_encoded(key),
                         limit,
                         &mut statistics,
+                        key_only,
+                    ).map_err(Error::from)
+                        .map(|r| {
+                            let mut valid_keys = 0;
+                            let mut bytes_read = 0;
+                            let mut stats = Statistics::default();
+                            r.iter().for_each(|r| {
+                                if let Ok(ref pair) = *r {
+                                    valid_keys += 1;
+                                    bytes_read += pair.0.len() + pair.1.len();
+                                }
+                            });
+                            stats.data.flow_stats.read_keys = valid_keys;
+                            stats.data.flow_stats.read_bytes = bytes_read;
+                            thread_ctx.collect_read_flow(ctx.get_region_id(), &stats);
+                            thread_ctx.collect_key_reads(CMD, valid_keys as u64);
+                            r
+                        });
+
+                    thread_ctx.collect_scan_count(CMD, &statistics);
+
+                    result
+                })
+                .then(move |r| {
+                    _timer.observe_duration();
+                    r
+                })
+        });
+
+        future::result(res)
+            .map_err(|_| Error::SchedTooBusy)
+            .flatten()
+    }
+
+    fn raw_batch_scan(
+        snapshot: Box<Snapshot>,
+        keys: Vec<Key>,
+        start_key_limit: usize,
```

Comment:
I prefer `limit`.

Reply:
Do you mean one limit for all start keys and dived quota for each start key evenly, am I right?

Reply:
After some thought, I guess making total limit the same as sum of all start key limits is reasonable. And I don't have any particular case that total limit should be smaller. So let me change it to one total limit.

Reply:
I think just a `key_limit` is better, and `total_limit` can be dropped, so you don't have to divide with key numbers.